File: app/src/main/java/com/example/dailyupdate/networking/GitHubService.java
package com.example.dailyupdate.networking;

import com.example.dailyupdate.data.model.GitHubResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GitHubService {

    @GET("search/repositories")
    Call<GitHubResponse> getGitHubRepoList(@Query("q") String searchKeyword,
                                           @Query("sort") String sortBy);

    @GET("search/repositories")
    Call<GitHubResponse> getGitHubRepoListWithOrder(@Query("q") String searchKeyword,
                                                    @Query("sort") String sortBy,
                                                    @Query("order") String searchOrder);
}

File: app/src/main/java/com/example/dailyupdate/networking/GitHubService.kt
package com.example.dailyupdate.networking

import com.example.dailyupdate.data.models.GitHubResponse
import retrofit2.http.GET
import retrofit2.http.Query

interface GitHubService {
    @GET("search/repositories")
    suspend fun getGitHubRepoList(@Query("q") searchKeyword: String,
                                  @Query("sort") sortBy: String,
                                  @Query("order") searchOrder: String): GitHubResponse
}